be/src/olap/tablet_manager.cpp

```diff
@@ -1272,11 +1185,19 @@ Status TabletManager::_create_tablet_meta_unlocked(const TCreateTabletReq& reque
     RETURN_NOT_OK_LOG(store->get_shard(&shard_id), "fail to get root path shard");
     Status res = TabletMeta::create(request, TabletUid::gen_uid(), shard_id, next_unique_id,
                                     col_idx_to_unique_id, tablet_meta);
-
-    if (request.__isset.storage_format && request.storage_format != TStorageFormat::V1) {
-        (*tablet_meta)->set_preferred_rowset_type(BETA_ROWSET);
-    } else {
-        (*tablet_meta)->set_preferred_rowset_type(ALPHA_ROWSET);
+    RETURN_NOT_OK(res);
+    if (request.__isset.storage_format) {
```

Comment:
check code in create_rowset_writer(), if storage_format is not set, preferred_rowset_type will be null, but it is used in create_rowset_writer(), it will make an unpredictable error.